packages/firestore/src/model/values.ts

```diff
@@ -385,6 +385,10 @@ function canonifyGeoPoint(geoPoint: api.LatLng): string {
   return `geo(${geoPoint.latitude},${geoPoint.longitude})`;
 }
 
+function canonifyReference(referenceValue: string): string {
+  return DocumentKey.fromName(referenceValue).toString();
```

Comment:
Doesn't this break existing canonical IDs?

Reply:
Yes, but the feature branch includes the schema migration.